### PR TITLE
Rename state props

### PR DESCRIPTION
**Pull request by clemp6r**
I modified the Store interface a little so it is more conventional with usual Kotlin design.

The `getState()` method is not appropriate because it is a getter, and in Kotlin we use properties instead. 

A property is a combination of a implicit getter and a backing field, so explicit getters like `getState()` are not required anymore. Also, there was a property named `state` so the Kotlin compiler would generate a `getState()` function for this property that clashes with the other `getState()` method you created. As a consequence, Java users would not be able to use your `getState()` because they generated `getState()` replaces it.

To summarize:
```
fun getState()  →  val state
val state       →  val stateChanges
```

### CODE REVIEW

**Comment by brianegan** [comment]
Ah, the naming :) This is something I  struggled with as well while trying to make this library idiomatic in Kotlin. I like the proposal to change the name from getState() to simply state. I should have done that from the get go, but used a convention we have here at SoundCloud in a Java context, and I don't think it's the right call for a public Kotlin project. 

I'm gonna think about the name for the state observable a little more, although stateChanges is quite nice, maybe even stateObservable since it's just so darn obvious.

File: bansa/src/main/kotlin/com/brianegan/bansa/Store.kt [hunk 1]
```diff
@@ -5,8 +5,8 @@ import rx.Subscriber
 import rx.Subscription
 
 interface Store<S, A> {
-    val state: Observable<S>
-    fun getState(): S
+    val stateChanges: Observable<S>
+    val state: S
```
Comment (brianegan) [inline bansa/src/main/kotlin/com/brianegan/bansa/Store.kt]: Neat Kotlin trick :) I like how you're keeping the interface read-only while using a `var` under the hood.

**Comment by brianegan** [comment]
Pondered over other names. Read through the code a few times. I like it! Let's go with it :)
